src/pages/costModels/components/paginationToolbarTemplate.tsx

```diff
@@ -0,0 +1,56 @@
+import {
+  Pagination,
+  PaginationProps,
+  Toolbar,
+  ToolbarContent,
+  ToolbarItem,
+  ToolbarProps,
+} from '@patternfly/react-core';
+import { TranslationFunction } from 'i18next';
+import React from 'react';
+import { withI18nProps } from './withI18nProps';
+
+type UniqueProps = Pick<ToolbarProps, 'id' | 'aria-label'>;
+type PaginationToolbarTemplateProps = PaginationProps & UniqueProps;
+
+export const PaginationToolbarTemplate: React.FunctionComponent<PaginationToolbarTemplateProps> = ({
+  id,
+  'aria-label': ariaLabel,
+  itemCount,
+  perPage,
+  page,
+  variant,
+  onPerPageSelect,
+  onSetPage,
+}) => {
+  return (
+    <Toolbar id={id}>
+      <ToolbarContent aria-label={ariaLabel}>
+        <ToolbarItem variant="pagination">
+          <Pagination
+            variant={variant}
+            itemCount={itemCount}
+            perPage={perPage}
+            page={page}
+            onSetPage={onSetPage}
+            onPerPageSelect={onPerPageSelect}
+          />
+        </ToolbarItem>
+      </ToolbarContent>
+    </Toolbar>
+  );
+};
+
+const translatePaginationToolbarProps = (
+  t: TranslationFunction,
+  props: PaginationToolbarTemplateProps
+) => {
+  return {
+    ...props,
+    'aria-label': t(props['aria-label']),
+  };
+};
+
+export default withI18nProps(translatePaginationToolbarProps)(
```

Comment:
It's not clear why `TranslationFunction` is used. Can we not use `translate()` here?

Reply:
I don't like `translate()` because it adds the `t` functions to my props which makes components coupled to it. In addition, running tests on the `translate` components fails or if I am wrapping it with the I18n provider it just does not render. However, if I am using the render-props version:
```
return (
  <I18n>{t => { ... }</I18n>
)
```

the tests does not fail and everything works fine.

That being said, I can discard `withI18nProps` in favor of render props. Will that make it clear?

Reply:
That's ok. Thanks for the explanation.

Reply:
Thanks. I updated the PR and discarded the `withI18nProps`.